sql/handler.h

```diff
@@ -1207,6 +1211,33 @@ struct Ha_fk_column_type {
   bool is_unsigned;
 };
 
+enum snapshot_operation {
+  SNAPSHOT_CREATE,
+  SNAPSHOT_ATTACH,
+  SNAPSHOT_RELEASE,
+  SNAPSHOT_NONE
+};
+
+struct snapshot_info_st {
+  std::string binlog_file;
+  ulonglong binlog_pos = 0;
+  std::string gtid_executed;
+  ulonglong snapshot_id = 0;
+  enum snapshot_operation op = snapshot_operation::SNAPSHOT_NONE;
```

Comment:
c-style enum declaration. Moreover `snapshot_operation` is not declared as `enum class`
```suggestion
  snapshot_operation op = SNAPSHOT_NONE;
```